loitering/create_raw_loitering/pipeline.py

```diff
@@ -52,22 +53,37 @@ def __init__(self, options):
         date_range = (start_date_with_buffer, end_date)
         labels = list_to_dict(gCloudParams.labels)
 
-        bq = bigquery.Client(project=gCloudParams.project)
-
-        # Ensure we delete any existing rows from the date to be processed.
-        # Needed to maintain consistency if are re-processing dates.
-        logger.info("Deleting existing rows with start_date {}".format(start_date))
-        bq.query(DELETE_QUERY.format(table=params.sink, start_date=start_date))
-
         (
             self.pipeline
             | ReadSource(date_range=date_range, source_table=params.source, labels=labels, source_timestamp_field=params.source_timestamp_field)
             | CalculateHourlyStats(slow_threshold=params.slow_threshold)
             | SlidingWindowByDay()
             | GroupLoiteringRanges(date_range=date_range)
             | CalculateLoiteringStats()
-            | WriteSink(sink_table=params.sink, description=get_description(params))
+            | WriteSink(sink_table=params.sink)
         )
 
+        self.params = params
+        self.gcloud_params = gCloudParams
+        self.start_date = start_date
+        self.end_date = end_date
+
     def run(self):
-        return self.pipeline.run()
+        bq = bigquery.Client(project=self.gcloud_params.project)
+
+        # Ensure we delete any existing rows from the date to be processed.
+        # Needed to maintain consistency if are re-processing dates.
+        logger.info("Deleting existing rows with start_date {}".format(self.start_date))
+        bq.query(DELETE_QUERY.format(table=self.params.sink, start_date=self.start_date))
+
+        result = self.pipeline.run()
+        result.wait_until_finish()
```

Comment:
I agree on having the `wait_until_finish()` here, that means that the option in the argument parser is not longer required